pilot/platform/cloudfoundry/servicediscovery_test.go

```diff
@@ -1,220 +0,0 @@
-
-	g.Expect(err).To(gomega.BeNil())
-	g.Expect(serviceModel).To(gomega.Equal(&model.Service{
-		Hostname: "process-guid-b.cfapps.internal",
-		Ports:    []*model.Port{{Port: 8080, Protocol: model.ProtocolTCP}},
-	}))
-}
-
-func TestServiceDiscovery_GetService_NotFound(t *testing.T) {
-	g := gomega.NewGomegaWithT(t)
-	state := newSDTestState()
-
-	state.mockClient.RoutesOutput.Ret0 <- makeSampleClientResponse()
-	state.mockClient.RoutesOutput.Ret1 <- nil
-
-	serviceModel, err := state.serviceDiscovery.GetService("does-not-exist.cfapps.internal")
-
-	g.Expect(err).To(gomega.BeNil())
-	g.Expect(serviceModel).To(gomega.BeNil())
-}
-
-func TestServiceDiscovery_GetService_ClientError(t *testing.T) {
-	g := gomega.NewGomegaWithT(t)
-	state := newSDTestState()
-
-	state.mockClient.RoutesOutput.Ret0 <- nil
-	state.mockClient.RoutesOutput.Ret1 <- errors.New("potato")
-
-	serviceModel, err := state.serviceDiscovery.GetService("process-guid-b.cfapps.internal")
-
-	g.Expect(err).To(gomega.MatchError("getting services: potato"))
-	g.Expect(serviceModel).To(gomega.BeNil())
-}
-
-func TestServiceDiscovery_Instances_Filtering(t *testing.T) {
-	g := gomega.NewGomegaWithT(t)
-	state := newSDTestState()
-
-	state.mockClient.RoutesOutput.Ret0 <- makeSampleClientResponse()
-	state.mockClient.RoutesOutput.Ret1 <- nil
-
-	instances, err := state.serviceDiscovery.Instances("process-guid-a.cfapps.internal", nil, nil)
-	g.Expect(err).To(gomega.BeNil())
-
-	servicePort := &model.Port{
-		Port:     8080,
-		Protocol: model.ProtocolTCP,
-	}
-	service := &model.Service{
-		Hostname: "process-guid-a.cfapps.internal",
-		Ports:    []*model.Port{servicePort},
-	}
-
-	g.Expect(instances).To(gomega.ConsistOf([]*model.ServiceInstance{
-		{
-			Endpoint: model.NetworkEndpoint{
-				Address:     "10.10.1.5",
-				Port:        61005,
-				ServicePort: servicePort,
-			},
-			Service: service,
-		},
-		{
-			Endpoint: model.NetworkEndpoint{
-				Address:     "10.0.40.2",
-				Port:        61008,
-				ServicePort: servicePort,
-			},
-			Service: service,
-		},
-	}))
-}
-
-func TestServiceDiscovery_Instances_NotFound(t *testing.T) {
-	g := gomega.NewGomegaWithT(t)
-	state := newSDTestState()
-
-	state.mockClient.RoutesOutput.Ret0 <- makeSampleClientResponse()
-	state.mockClient.RoutesOutput.Ret1 <- nil
-
-	instances, err := state.serviceDiscovery.Instances("non-existent.cfapps.internal", nil, nil)
-	g.Expect(err).To(gomega.BeNil())
-	g.Expect(instances).To(gomega.BeNil())
-}
-
-func TestServiceDiscovery_Instances_ClientError(t *testing.T) {
-	g := gomega.NewGomegaWithT(t)
-	state := newSDTestState()
-
-	state.mockClient.RoutesOutput.Ret0 <- nil
-	state.mockClient.RoutesOutput.Ret1 <- errors.New("potato")
-
-	serviceModel, err := state.serviceDiscovery.Instances("process-guid-b.cfapps.internal", nil, nil)
-
-	g.Expect(err).To(gomega.MatchError("getting instances: potato"))
-	g.Expect(serviceModel).To(gomega.BeNil())
-}
```

Comment:
And since you are touching CloudFoundry code, it would be nice to get @rosenhouse or some one from pivotal to ensure that this does not break their existing integration. Sorry for the trouble, but this is needed until we get the full CF tests into pilot.

Reply:
notwithstanding other comments above, this looks ok from a CF perspective.  We've started on basic integration tests though, https://github.com/istio/istio/issues/2204 . We will update this issue once we have something useful.